### DISCUSSION TITLE
print on UART0

### DISCUSSION THREAD

**Discussion by CrabbyPete** [micropython/micropython#14014] [RP2040 / Pico] [answered]
I am using Thonny to program my Raspberry Pi Pico in python, and I have print statements to monitor the code. Once its complete I want to be able to connect a battery to the pico and monitor the output of those print statements on UART0. I have and FTDI cable on UART0, but I don't see any output. Is the USB port still active? How can I make the print statements go to the UART0 pins?

**Comment by robert-hh** [comment, marked as answer]
You can use os.dupterm() to replicate the standard output to a stream device, like UART. e.g.

```
import os
from machine import UART
uart=UART(0, 115200)
os.dupterm(uart, 0)
```

**Reply by GitHubsSilverBullet** [reply]
Maybe something like:
```python
u = machine.UART(0,9600,8,None,1)
print('testing', file=u)
```
